src/terminal/adapter/ut_adapter/adapterTest.cpp

```diff
@@ -2263,6 +2263,141 @@ class AdapterTest
+        expectedResponse += L"9;1;120;50;100/";
+        // RGB(0,255,0) -> HLS(240°,50%,100%)
+        expectedResponse += L"10;1;240;50;100/";
+        // RGB(255,255,0) -> HLS(180°,50%,100%)
+        expectedResponse += L"11;1;180;50;100/";
+        // RGB(0,0,255) -> HLS(0°,50%,100%)
+        expectedResponse += L"12;1;0;50;100/";
+        // RGB(255,0,255) -> HLS(60°,50%,100%)
+        expectedResponse += L"13;1;60;50;100/";
+        // RGB(0,255,255) -> HLS(300°,50%,100%)
+        expectedResponse += L"14;1;300;50;100/";
+        // RGB(255,255,255) -> HLS(0°,100%,0%)
+        expectedResponse += L"15;1;0;100;0/";
+        // Remaining slots are black, i.e. HLS(0°,0%,0%)
+        for (size_t i = 16; i < TextColor::TABLE_SIZE; i++)
+        {
+            expectedResponse += std::to_wstring(i) + L";1;0;0;0";
+            if (i + 1 < TextColor::TABLE_SIZE)
+            {
+                expectedResponse += L'/';
+            }
+        }
+        expectedResponse += ST;
+        _testGetSet->ValidateInputEvent(expectedResponse.c_str());
+
+        Log::Comment(L"RGB color model");
+
+        const auto rgbColorModel = static_cast<int>(DispatchTypes::ColorModel::RGB);
+        _pDispatch->RequestTerminalStateReport(DispatchTypes::ReportFormat::ColorTableReport, rgbColorModel);
+
+        expectedResponse = DECCTR;
+        // RGB(0,0,0) -> RGB(0%,0%,0%)
+        expectedResponse += L"0;2;0;0;0/";
+        // RGB(204,36,36) -> RGB(80%,14%,14%)
+        expectedResponse += L"1;2;80;14;14/";
+        // RGB(51,204,51) -> RGB(20%,80%,20%)
+        expectedResponse += L"2;2;20;80;20/";
+        // RGB(204,204,51) -> RGB(80%,80%,20%)
+        expectedResponse += L"3;2;80;80;20/";
+        // RGB(51,51,204) -> RGB(20%,20%,80%)
+        expectedResponse += L"4;2;20;20;80/";
+        // RGB(204,51,204) -> RGB(80%,20%,80%)
+        expectedResponse += L"5;2;80;20;80/";
+        // RGB(51,204,204) -> RGB(20%,80%,80%)
+        expectedResponse += L"6;2;20;80;80/";
+        // RGB(120,120,120) -> RGB(47%,47%,47%)
+        expectedResponse += L"7;2;47;47;47/";
+        // RGB(69,69,69) -> RGB(27%,27%,27%)
+        expectedResponse += L"8;2;27;27;27/";
+        // RGB(255,0,0) -> RGB(100%,0%,0%)
+        expectedResponse += L"9;2;100;0;0/";
+        // RGB(0,255,0) -> RGB(0%,100%,0%)
+        expectedResponse += L"10;2;0;100;0/";
+        // RGB(255,255,0) -> RGB(100%,100%,0%)
+        expectedResponse += L"11;2;100;100;0/";
+        // RGB(0,0,255) -> RGB(0%,0%,100%)
+        expectedResponse += L"12;2;0;0;100/";
+        // RGB(255,0,255) -> RGB(100%,0%,100%)
+        expectedResponse += L"13;2;100;0;100/";
+        // RGB(0,255,255) -> RGB(0%,100%,100%)
+        expectedResponse += L"14;2;0;100;100/";
+        // RGB(255,255,255) -> RGB(100%,100%,100%)
+        expectedResponse += L"15;2;100;100;100/";
+        // Remaining slots are black, i.e. RGB(0%,0%,0%)
+        for (size_t i = 16; i < TextColor::TABLE_SIZE; i++)

```

Comment:
for `TABLE_SIZE`: is it correct that we will emit the `FRAME_BACKGROUND` et al? Are those indices stable across terminal emulators?

Reply:
what if they are set to `INVALID_COLOR`?

Reply:
Of the terminals that support this, there is one which is a genuine terminal emulator, which only returns 16 colors (since that is what you would get on the original hardware terminals). The others that I know of return 256 colors, which includes the color cube and gray scale entries that are common to most modern terminals.

I wanted to include our entire color table, even though the indices aren't standard, because you shouldn't really need to care about that - it just provides a way to save and restore the entire set. However, I think the `INVALID_COLOR` entries are going to be a problem, because they're not going to restore correctly, and that could break things.

But maybe we can just drop any colors from the report that are `INVALID_COLOR`, and that way there's no risk of corrupting them with a simple save and restore that doesn't change anything. We won't then be able to restore any `INVALID_COLOR` entries that have been changed, but that's no worse than if we only reported the first 256.